source/common/network/address_impl.h

```diff
@@ -68,6 +68,7 @@ class Ipv4Instance : public InstanceBase {
 
   struct IpHelper : public Ip {
     const std::string& addressAsString() const override { return friendly_address_; }
+    bool isAnyAddress() const override { return friendly_address_ == "0.0.0.0"; }
```

Comment:
Instead of a string match here, can we actually match on the underlying bytes? E.g., IPADDR_ANY for IPv4, and whatever the equiv is for IPv6.